Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_runner.py

```diff
@@ -570,6 +571,21 @@ def __eq__(self, other):
 
 
 class Sharder(object):
+    MAX_SIZE = 128
+
+    @classmethod
+    def shards_for_directory(cls, directory, test_inputs, depth=0):
+        if len(test_inputs) == 0:
+            return []
+        if len(test_inputs) < cls.MAX_SIZE or depth > 31:
+            return [TestShard(directory, test_inputs)]
+        mask = 1 << depth
+        split_inputs = ([], [])
```

Comment:
I guess this makes sense, but it's kinda surprising that we have only have a two-way split here, given we end up with an arbitrary number of shards.